saidump/saidump.cpp

```diff
@@ -399,7 +416,154 @@ void dumpGraph(const TableDump& td)
+    {        
+        SWSS_LOG_ERROR_AND_STDERR("Get %s's size failure or its size %ld  >= %ld MB.", file_name.c_str(), file_size, RDB_FILE_MAX_SIZE / 1024 / 1024);
+        return SAI_STATUS_FAILURE;
+    }
+
+    input_file.seekg(0);     // Move to the begin of the file
+
+    // Read the content of the input file into a string
+    std::string content((std::istreambuf_iterator<char>(input_file)),
+                   std::istreambuf_iterator<char>());
+
+    content = regex_replace(content, std::regex("\\},\\{\\r"), ",");
+
+    std::ofstream outputFile(file_name);
+
+    if (!outputFile.is_open())
+    {
+        SWSS_LOG_ERROR_AND_STDERR("Failed to open the output file %s.", file_name.c_str());
+        return SAI_STATUS_FAILURE;
+    }
+
+    //Romove the 1st and last char to make sure its format is same as previous saidump's output
+    if (content.size() >= 2 && content[0] == '[' && content[content.length()-1] == ']')
+    {
+        outputFile << content.substr(1, content.size()-2);
+    }
+    else
+    {
+        outputFile << content;
+    }
+
+    return SAI_STATUS_SUCCESS;
+}
+
+static sai_status_t dumpFromRedisRdbFile(std::string file_name)
+{
+    SWSS_LOG_ENTER();
+
+    std::ifstream input_file(file_name);
+
+    if (!input_file.is_open())
+    {
+        SWSS_LOG_ERROR_AND_STDERR("The file %s does not exist for dumping from redis dump.rdb file.", file_name.c_str());
+        return SAI_STATUS_FAILURE;
+    }
+
+    try
+    {
+        // Parse the JSON data from the file (validation)
+        nlohmann::json jsonData;
+        input_file >> jsonData;
+
+        SWSS_LOG_DEBUG("JSON file is valid.");
+
+        for (json::iterator it = jsonData.begin(); it != jsonData.end(); ++it)
+        {
+            json jj_key = it.key();
+
+            std::string keystr = jj_key;
+            std::string item_name = keystr;
+            size_t pos = keystr.find_first_of(":");
+
+            if (pos != std::string::npos)
+            {
+                if(ASIC_STATE_TABLE != keystr.substr(0, pos))  // filter out non ASIC_STATE
+                {
+                    continue;
+                }
+
+                item_name = keystr.substr(pos + 1);
+
+                if (item_name.find(":") != std::string::npos)
+                {
+                    item_name.replace(item_name.find_first_of(":"), 1, " ");
+                }
+            }
+            else
+            {
+                continue;
+            }
+
+            std::cout << item_name << " " << std::endl;
+
+            json jj = it.value();
+
+            if (!it->is_object())
+            {
+                continue;
+            }
+
+            TableMap map;
+
+            for (json::iterator itt = jj.begin(); itt != jj.end(); ++itt)
+            {
+                if (itt.key() == "NULL")
+                {
+                    continue;
+                }
+
+                map[itt.key()] = itt.value();
+            }
+
+            size_t indent = 4;
```

Comment:
make this const expr and it could be LINE_IDENT etc

Reply:
Fix it.